include/devicetree/zephyr.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Zephyr-specific devicetree /chosen properties
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_ZEPHYR_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_ZEPHYR_H_
+
+/**
+ * @defgroup devicetree-zephyr Zephyr's /chosen nodes
+ * @{
+ */
+
+/*
+ * This file is currently deliberately not defining macros for some
+ * existing zephyr,foo chosen nodes, such as zephyr,sram, until there
+ * are users for them. Feel free to extend it as needed.
+ */
+
+#if DT_HAS_CHOSEN(zephyr_flash)
+/** Node identifier for the /chosen node's zephyr,flash property value. */
+#define DT_ZEPHYR_FLASH DT_CHOSEN(zephyr_flash)
```

Comment:
Feel like putting CHOSEN in the define name.. so either `DT_CHOSEN_ZEPHYR_FLASH` (which would map to the path) or `DT_ZEPHYR_CHOSEN_FLASH`

Reply:
DT_CHOSEN_ZEPHYR_FLASH isn't any shorter than DT_CHOSEN(zephyr_flash) and neither is really needed, so I've removed this macro and am focusing on the label.